### DISCUSSION TITLE
cloudflare and nextcloud-aio

### DISCUSSION THREAD

**Discussion by KleIKKK** [nextcloud/all-in-one#757] [Questions] [answered]
Hello guys,

I am trying to host my own Nextcloud on my local machine that has no public IP (since it's behind NAT and I got no access to router).
So based on some research I did on the internet, I went with Cloudflare tunnel. I set up a tunnel inside a docker. I made sure it does work, because when I pass as a port 9000, my website points to my Portainer, but now I am trying, to make it work for nextcloud-aio. But I cannot figure out how I should approach it.

command that I create NC-Aio:
```
sudo docker run -d -it -e NEXTCLOUD_DATADIR="/home/nuc/nextcloud" -e NEXTCLOUD_MOUNT="/mnt/" --name nextcloud-aio-mastercontainer --restart always -p 8080:8080 -e APACHE_PORT=11000 --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config --volume /var/run/docker.sock:/var/run/docker.sock:ro nextcloud/all-in-one:latest
```   

Inside Cloudflare tunnel, I got below settings:
![image](https://user-images.githubusercontent.com/42144903/170844453-82db6750-fc93-4714-aa95-03cc359fae7c.png)

then when I go to over my local network to below website I can see a problem:
![image](https://user-images.githubusercontent.com/42144903/170844471-696589ce-9a1f-4c3a-a863-b272e2a92cfb.png)

I also tried to change inside Cloudflare settings from HTTPS to HTTP, but then I got below error:

![image](https://user-images.githubusercontent.com/42144903/170844493-da8f8472-dd52-4291-a5ef-ce3a57bacf15.png)

Please, if somebody could help me out, that would be awesome.

**Comment by szaimen** [comment, marked as answer]
Hello, I just added documentation and a flag with 1.5.0 Beta that allows to run AIO behind a Cloudflare tunnel. See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel how to switch the channel and https://github.com/nextcloud/all-in-one#how-to-run-it-behind-a-cloudflare-argo-tunnel. 
Feedback is welcome!
